src/geode/mesh/builder/solid_mesh_builder.cpp

```diff
@@ -463,78 +463,36 @@ namespace geode
     void SolidMeshBuilder< dimension >::compute_polyhedron_adjacencies(
         absl::Span< const index_t > polyhedra_to_connect )
     {
-        absl::FixedArray< absl::InlinedVector< PolyhedronFacet, 4 > >
-            polyhedron_facet_around_vertex( solid_mesh_->nb_vertices() );
+        const auto facets_enabled = solid_mesh_->are_facets_enabled();
+        solid_mesh_->enable_facets();
+        const auto& facets = solid_mesh_->facets();
+        absl::FixedArray< absl::InlinedVector< PolyhedronFacet, 2 > >
+            polyhedron_facets( facets.nb_facets() );
         for( const auto polyhedron : polyhedra_to_connect )
         {
             for( const auto f :
                 Range{ solid_mesh_->nb_polyhedron_facets( polyhedron ) } )
             {
-                const PolyhedronFacet facet{ polyhedron, f };
-                for( const auto v : Range{
-                         solid_mesh_->nb_polyhedron_facet_vertices( facet ) } )
-                {
-                    const auto vertex =
-                        solid_mesh_->polyhedron_facet_vertex( { facet, v } );
-                    polyhedron_facet_around_vertex[vertex].push_back( facet );
-                }
+                PolyhedronFacet facet{ polyhedron, f };
+                const auto vertices_id =
+                    solid_mesh_->polyhedron_facet_vertices( facet );
+                const auto facet_id = facets.facet_from_vertices( vertices_id );
+                polyhedron_facets[facet_id.value()].emplace_back(
+                    std::move( facet ) );
```

Comment:
Is it not faster to use a `absl::FixedArray< PolyhedronFacet >` instead of `absl::FixedArray< absl::InlinedVector< PolyhedronFacet, 2 > >` and if there is already the facet you set the adjacency?

Reply:
And lighter?